src/hp/derp/client.rs

```diff
@@ -591,3 +556,105 @@ impl RateLimiter {
         }
     }
 }
+
+async fn send_packet<W: AsyncWrite + Unpin>(
+    mut writer: W,
+    rate_limiter: &Option<RateLimiter>,
+    dstkey: PublicKey,
+    packet: &[u8],
+) -> Result<()> {
+    if packet.len() > MAX_PACKET_SIZE {
+        bail!("packet too big: {}", packet.len());
+    }
+    let frame_len = PUBLIC_KEY_LENGTH + packet.len();
+    if let Some(rate_limiter) = rate_limiter {
+        match rate_limiter.check_n(frame_len) {
+            Ok(_) => {}
+            Err(_) => {
+                tracing::warn!("dropping send: rate limit reached");
+                return Ok(());
```

Comment:
`if rate_limiter.check_n(frame_len).is_err() { .. }` would be a bit more straightforward to read

Reply:
🔥